virttest/qemu_devices/qcontainer.py

```diff
@@ -131,19 +131,19 @@ def get_qmp_cmds(qemu_binary, workaround_qemu_qmp_crash=False):
         self.__qemu_binary = qemu_binary
         self.__execute_qemu_last = None
         self.__execute_qemu_out = ""
-        # Check whether we need to add machine_type
-        cmd = "%s -device \? 2>&1" % qemu_binary
+        # Check whether we need to roll back to tcg mode
+        cmd = "echo -e 'quit' | %s -monitor stdio -vnc none -S" % qemu_binary
         result = process.run(cmd, timeout=10,
                              ignore_status=True,
+                             allow_output_check="combined",
                              shell=True,
                              verbose=False)
-        # Some architectures (arm) require machine type to be always set
-        if result.exit_status and b"machine specified" in result.stdout:
-            self.__workaround_machine_type = True
-            basic_qemu_cmd = "%s -machine virt" % qemu_binary
+        # Some architectures (arm) require machine/cpu type to be always set
+        # to launch a kvm instance
+        if result.exit_status and "kvm_init_vcpu failed" in result.stdout_text:
+            self.__basic_qemu_cmd = "%s -accel tcg" % qemu_binary
```

Comment:
Can it work properly with using `-machine none`?

Reply:
Updated.